### PR TITLE
Add connection to PM's AWS RDS instance

### PR DESCRIPTION
**Pull request by mzappitello**
* Add ssl cert to docker container using curl
* update postgres utils to connect to the remote rds if unable to find a password in the environment variables
* Add logging, but not re throwing if unable to find .env file in startup.
* Move command for running startup.py from docker-compose to dockerfile.
* Move verbose logging of sqlalchemy into cli args.
* Move db seeding to DBManager constructor from main method of startup dot py.

Asana Task: <asana_ticket_url>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -7,7 +7,7 @@ ENV PYTHONUNBUFFERED 1
 
 # Install dev dependencies
 RUN apt-get update
-RUN apt-get install -y libpq-dev gcc
```
Comment (mzappitello) [inline Dockerfile:10]: need curl for getting the ssl cert.

File: Dockerfile [hunk 2]
```diff
@@ -17,3 +17,10 @@ COPY ./performance_manager /performance_manager/
 WORKDIR /performance_manager/
 RUN poetry config virtualenvs.create false
 RUN poetry install --no-dev --no-interaction --no-ansi
+
+# Fetch Amazon RDS certificate chain
+RUN curl https://truststore.pki.rds.amazonaws.com/global/global-bundle.pem -o /usr/local/share/amazon-certs.pem
+RUN echo "acaf8712f8d71c05f85503c6b90fd0127e95ff0091bf094a22a650119684a08e /usr/local/share/amazon-certs.pem" | sha256sum -c -
+RUN chmod a=r /usr/local/share/amazon-certs.pem
```
Comment (mzappitello) [inline Dockerfile:24]: get the ssl cert so we can use it to connect to the RDS.

File: Dockerfile [hunk 3]
```diff
@@ -17,3 +17,10 @@ COPY ./performance_manager /performance_manager/
...
+
+CMD ["python", "startup.py"]
```
Comment (mzappitello) [inline Dockerfile:26]: start it up on launch.

File: docker-compose.yml [hunk 4]
```diff
@@ -25,4 +25,3 @@ services:
     working_dir: /performance_manager
     volumes:
       - ~/.aws:/root/.aws:ro # map credentials to be used by boto3, read-only
-    command: ["python", "startup.py"]
```
Comment (mzappitello) [inline docker-compose.yml:28]: moved this to the dockerfile itself.

File: performance_manager/lib/postgres_utils.py [hunk 5]
```diff
@@ -25,24 +26,61 @@ def get_local_engine(
...
+        db_ssl_options = ""
+
+        assert db_host is not None
+        assert db_name is not None
+        assert db_port is not None
+        assert db_user is not None
 
         process_logger.add_metadata(
-            host=host, database_name=dbname, user=user, port=port
+            host=db_host, database_name=db_name, user=db_user, port=db_port
```
Comment (mzappitello) [inline performance_manager/lib/postgres_utils.py:43]: some slight renaming and using `get` here.

setting us up to get the AWS url if a password isn't provided (just like we do in the gtfs ingestion).

File: performance_manager/lib/postgres_utils.py [hunk 6]
```diff
@@ -25,24 +26,61 @@ def get_local_engine(
...
+            # set the ssl cert path to the file that should be added to the
+            # lambda function at deploy time
+            db_ssl_cert = os.path.abspath(
+                os.path.join("/", "usr", "local", "share", "amazon-certs.pem")
+            )
+
+            assert os.path.isfile(db_ssl_cert)
+
+            # update the ssl options string to add to the database url
+            db_ssl_options = f"?sslmode=verify-full&sslrootcert={db_ssl_cert}"
```
Comment (mzappitello) [inline performance_manager/lib/postgres_utils.py:76]: get the aws password via iam and setup ssl options.

File: performance_manager/lib/postgres_utils.py [hunk 7]
```diff
@@ -25,24 +26,61 @@ def get_local_engine(
...
+
+        database_url = (
+            f"postgresql+psycopg2://{db_user}:"
+            f"{db_password}@{db_host}/{db_name}"
+            f"{db_ssl_options}"
         )
+
```
Comment (mzappitello) [inline performance_manager/lib/postgres_utils.py:83]: create the db url from the pieces.

File: performance_manager/lib/postgres_utils.py [hunk 8]
```diff
@@ -112,6 +155,9 @@ def set_sqlite_pragma(dbapi_connection, _):  # type: ignore
         # create tables in SqlBase
         SqlBase.metadata.create_all(self.engine)
 
+        if seed:
+            self.seed_metadata()
+
```
Comment (mzappitello) [inline performance_manager/lib/postgres_utils.py:160]: moved seeding into constructor, taking it out of `main`

File: performance_manager/startup.py [hunk 9]
```diff
@@ -45,6 +45,8 @@ def load_environment() -> None:
                 logging.info("setting %s to %s", key, value)
                 os.environ[key] = value
 
+    except FileNotFoundError as fnfe:
+        logging.warning("unable to find env file %s", fnfe)
```
Comment (mzappitello) [inline performance_manager/startup.py:49]: don't throw if we can't find the file, just warn that the env hasn't been updated.

**Comment by rymarczy** [comment]
LGTM

**Pull request merged**